src/app/CommandHandler.cpp

```diff
@@ -180,63 +189,78 @@ CHIP_ERROR CommandHandler::SendCommandResponse()
 
 CHIP_ERROR CommandHandler::ProcessCommandDataIB(CommandDataIB::Parser & aCommandElement)
 {
-    CHIP_ERROR err = CHIP_NO_ERROR;
+    CHIP_ERROR err                      = CHIP_NO_ERROR;
+    SubjectDescriptor subjectDescriptor;
     CommandPathIB::Parser commandPath;
     TLV::TLVReader commandDataReader;
-    ClusterId clusterId;
+    RequestPath requestPath;
     CommandId commandId;
-    EndpointId endpointId;
+    Privilege privilege = Privilege::kOperate; // TODO: get custom required privilege using commandId
+
+    if (mpExchangeCtx && mpExchangeCtx->HasSessionHandle())
+    {
+        mpExchangeCtx->GetSessionHandle().GetSubjectDescriptor();
```

Comment:
Yeah that was the crashing test. Eventually I'll rewrite this to be more along the lines of the work that Boris is doing with fabric index. (And actually that's how I did it in earlier versions of this draft PR.) Right now it just needs to build and get in (to avoid infinite merge conflicts).